### PR TITLE
Make matmul tests take m,n,k directly 

### PR DESCRIPTION
**Pull request by newling**
@nirvedhmeshram I've been wanting to try this for a while, and your comment in the standup this morning made me just do it. 

If I understood correctly (sorry I was having audio issues) when you compile 2+ matmuls together, you hit runtime issues? 

This PR does:
1) removes the groups 'large' 'small' 'legacy_*' and now a test must be specified directly with m,n,k values. 
2) some minor superficial changes.

### CODE REVIEW

File: runtime/src/iree-amd-aie/driver/xrt/direct_command_buffer.cc [hunk 1]
```diff
@@ -353,6 +353,12 @@ static iree_status_t iree_hal_xrt_direct_command_buffer_dispatch(
           xrt::bo(*command_buffer->descriptor_sets[i].bindings[j],
                   command_buffer->descriptor_sets[i].lengths[j],
                   command_buffer->descriptor_sets[i].offsets[j]);
+
+      // TODO(jn) remove from PR. This is a temporary fix to avoid a
```
Comment (nirvedhmeshram) [inline runtime/src/iree-amd-aie/driver/xrt/direct_command_buffer.cc:357]: remove?

**Review by nirvedhmeshram — APPROVED** [review]
There is a trade-off here that we are taking the pressure away from runtime to be able to actually handle multiple dispatches at once. At the same time this allows us more flexibility in running different dtypes and shapes which is more important for the development phase we are in so lets do it this way. Thanks for the change!

**Comment by newling** [comment]
> There is a trade-off here that we are taking the pressure away from runtime to be able to actually handle multiple dispatches at once.

That was my thinking too. Hmm. I guess I could keep the option to run a group of shapes too, and have a test commented out with a "TODO: resolve runtime issues". Because another advantage of having multiple matmuls in the vmfb might be test time? I'll play around with this option, and if this PR changes dramatically I'll ask for another review before landing.

**Comment by nirvedhmeshram** [comment]
> > There is a trade-off here that we are taking the pressure away from runtime to be able to actually handle multiple dispatches at once.
> 
> That was my thinking too. Hmm. I guess I could keep the option to run a group of shapes too, and have a test commented out with a "TODO: resolve runtime issues". Because another advantage of having multiple matmuls in the vmfb might be test time? I'll play around with this option, and if this PR changes dramatically I'll ask for another review before landing.

Thanks, thats a good plan!

**Comment by newling** [comment]
Hey @nirvedhmeshram I've updated this to accept multiple matmul shapes in a single test. I've also added options for the case where we (1) just want to compile and (2) expect a compilation failure, to better capture the cases commented out currently. I'm going to be away for 8 days: please feel free to land or change this if you want. Thanks!

**Review by nirvedhmeshram — APPROVED** [review]
LGTM

**Pull request merged**